React/Nextjs Project Post Setup Steps.md

```diff
@@ -0,0 +1,679 @@
+                "prettier --write"
+            ]
+        }
+    }
+    ```
+   Note for `pre-push` hook: Tests configuration is required in order to run `npm test` on `pre-push`. This will be done later.
+1. Based on your IDE, some changes might be required in IDE configurations
+    1. VS Code
+        - Create file `settings.json` in `.vscode` folder with following content
+        ```json
+        {
+            "editor.defaultFormatter": "esbenp.prettier-vscode",
+            "editor.formatOnSave": true,
+            "editor.insertSpaces": false,
+            "[scss]": {
+                "editor.formatOnSave": false
+            }
+        }
+        ```
+       - Commit and push changes
+   1. Intellij IDEA / WebStorm
+    - In `preferences` on `macOS` or `settings` on `Windows`, enter search for `Prettier` and make setup for project
+        - `Prettier package` path should be set to `[project_path]/node_modules/prettier`
+        - `On save` option should be checked
+    - In `preferences` on `macOS` or `settings` on `Windows`, enter search for `ESLint` and make setup for project
+        - `Manual ESLint configuration ` option should be checked and path should be set to `[project_path]/node_modules/eslint`
+        - `Configuration File:` path should be set to `[project_path]/.eslintrc`
+        - `Run eslint —fix on save` option should be checked
+   - Commit and push these changes
+
+##### References:
+[Code quality](/books/frontend/javascript/code-quality)
+
+[EditorConfig](https://editorconfig.org/)
+
+
+### TypeScript
+
+##### Depends on sections: none
+
+| Package | Version |
+|---|---:|
+| typescript | 4.0.x |
+
+To add `TypeScript` support do the following:
+
+1. In project root create a new file with a name `tsconfig.json` (manually or via terminal).
+    ```bash
+    touch tsconfig.json
+    ```
+1.  Run command
+    ```bash
+    yarn add --dev typescript @types/react @types/node
+    ```
+    or
+    ```bash
+    npm install --save-dev typescript @types/react @types/node
+    ```
+    to install required packages.
+1. Run command
+    ```bash
+    yarn dev
+    ```
+    or
+    ```bash
+    npm run dev
+    ```
+    and `Next.js` will populate `tsconfig.json` file with default values.
+    
+    Edit these values or add new based on the project requirements.
+    ```bash
+    // add these additional rules in `compilerOptions` property
+    "experimentalDecorators": true,
+    "baseUrl": "src"
+    ```
+1. In the app rename all `js/jsx` files to `ts/tsx` files.
+1. Start adding types.
+    ```tsx
+   // _app.tsx
+    import { AppProps } from 'next/app';
+    
+    function MyApp({ Component, pageProps }: AppProps): ReactElement { }
+    ```
+   
+##### References:
+[TypeScript](https://www.typescriptlang.org/)
+
+[TypeScript in Next.js](https://nextjs.org/docs/basic-features/typescript)
+
+
+### Localisation
+
+##### Depends on sections: none
+
+| Package | Version |
+|---|---:|
+| i18next | 19.7.x |
```

Comment:
Does it make sense to define minor versions here or should we assume that the latest minor should be used?